src/kOS/Utilities/SteeringHelper.cs

```diff
@@ -162,31 +167,65 @@ public static Vector3d GetTorque(Vessel vessel, float thrust)
+                    yaw += Math.Abs(pod.rotPower);
                     roll += Math.Abs(pod.rotPower);
                 }
-
+                /*
+                //This is probably useless - commented out.
                 var rcsModule = part as RCSModule;
                 if (rcsModule != null)
                 {
                     float max = rcsModule.thrusterPowers.Aggregate<float, float>(0, Mathf.Max);
 
-                    pitchYaw += max * relCoM.magnitude;
+                    roll += max * (Vector3.Cross(relCoM, rollaxis)).magnitude;
+                    pitchYaw += max * Vector3.Dot(relCoM, rollaxis);
                 }
-
+                */
                 foreach (PartModule module in part.Modules)
                 {
                     var wheel = module as ModuleReactionWheel;
                     if (wheel == null) continue;
 
-                    pitchYaw += wheel.PitchTorque;
+                    pitch += wheel.PitchTorque;
+                    yaw += wheel.YawTorque;
                     roll += wheel.RollTorque;
                 }
-
-                pitchYaw += (float)GetThrustTorque(part, vessel) * thrust;
+                if (vessel.ActionGroups [KSPActionGroup.RCS])
+                {
+                    foreach (PartModule module in part.Modules) {
+                        var rcs = module as ModuleRCS;
+                        if (rcs == null) continue;
+
+                        if (rcs.rcsEnabled == true)
+                        {
+                            bool enoughfuel = true;
+                            foreach (Propellant p in rcs.propellants)
+                            {
+                                if ((int)(p.totalResourceAvailable)==0)
+                                {
+                                    enoughfuel = false;
+                                }
+                            }
+                            if (enoughfuel == true)
+                            {
+                                foreach (Transform thrustdir in rcs.thrusterTransforms)
+                                {
+                                    float rcsthrust = rcs.thrusterPower;
+                                    //just counting positive contributions in one direction. This is incorrect for asymmetric thruster placements.
+                                    roll += Mathf.Max(rcsthrust * Vector3.Dot(Vector3.Cross(relCoM, thrustdir.up), rollaxis), 0.0f);
+                                    pitch += Mathf.Max(rcsthrust * Vector3.Dot(Vector3.Cross(Vector3.Cross(relCoM, thrustdir.up), rollaxis), pitchaxis), 0.0f);
+                                    yaw += Mathf.Max(rcsthrust * Vector3.Dot(Vector3.Cross(Vector3.Cross(relCoM, thrustdir.up), rollaxis), Vector3.Cross(rollaxis,pitchaxis)),0.0f);
+                                }
+                            }
+                        }
+                    }
+                }
+                pitch += (float)GetThrustTorque(part, vessel) * thrust;
```

Comment:
GetThrustTorque doesnt actually work because it uses the engine derived classes that are not being used anymore. We need to get a new implementation that uses the gimbal module.